docs/docs/configuration/providers/ms_entra_id.md

```diff
@@ -0,0 +1,199 @@
+---
+id: ms_entra_id
+title: Microsoft Entra ID
+---
+OIDC-compliant provider for Microsoft Entra ID (Azure AD successor) application registrations. This providers support Azure oAuth V2 endpoint only, if you want to use V1 endpoint, try legacy `azure` provider. 
+
+## Configure application registration
+To start, create an App registration with minimal permissions, assign redirect URI and generate secret. All account types are supported (Single tenant, multi tenant, multi tenant with MS accounts, MS accounts only).
+<details>
+    <summary>See Azure Portal example</summary>
+    <div class="videoBlock">
+        <video controls>
+            <source src="/oauth2-proxy/azure-videos/create-app-reg.mkv"/>
+        </video>
+    </div>
+
+When created with Portal, App registration automatically creates a delegated API permission for `User.Read`. 
+</details>
+
+<details>
+    <summary>See Terraform example</summary>
+
+    resource "azuread_application" "auth" {
+        display_name     = "oauth2-proxy"
+        sign_in_audience = "AzureADMyOrg" # Others are also supported
+
+        web {
+            redirect_uris = [
+                "https://podinfo.lakis.tech/oauth2/callback",
+            ]
+        }
+
+        required_resource_access {
+            resource_app_id = "00000003-0000-0000-c000-000000000000" # Microsoft Graph
+            resource_access {
+                id   = "e1fe6dd8-ba31-4d61-89e7-88639da4683d" # User.Read delegated
+                type = "Scope"
+            }
+        }
+    }
+
+    resource "azuread_application_password" "apppass" {
+        application_id = azuread_application.auth.id
+    }
```

Comment:
Does it make sense to put this in a code block? (\```...```)

Reply:
Right, by some reason docusaurus misbehaved before, with code block everything looks good
![image](https://github.com/oauth2-proxy/oauth2-proxy/assets/12859079/150e247c-ba58-4ff0-8032-a0aa3279c502)